### PR TITLE
auto update subject when trainee update task within this subject

### PR DESCRIPTION
**Pull request by duypn-2245**

### CODE REVIEW

**Comment by duypn-2245** [comment]
ready

File: app/models/process_task.rb [hunk 1]
```diff
@@ -1,6 +1,20 @@
...
   belongs_to :user
   belongs_to :task
   has_many :histories, dependent: :destroy
+
+  scope :count_finished, ->(task_ids){where "status = ? and task_id IN (?)", ProcessTask.statuses[:finished], task_ids}
+
+  def auto_update_user_subject
+    f_task = task
+    f_subject = f_task.subject
+    f_task_ids = f_subject.tasks.ids
```
Comment (vanvtt-0952) [inline app/models/process_task.rb:13]: scope tìm kiềm task_ids với dk subject_id = task.subject_id

File: app/models/process_task.rb [hunk 2]
```diff
@@ -1,6 +1,20 @@
...
+    f_user_subject = UserSubject.find_by subject_id: f_subject.id, user_id: user_id
+    count_process_tasks = f_task_ids.size
+    count_process_tasks_finish = ProcessTask.count_finished(f_task_ids).size
+    count_process_tasks == count_process_tasks_finish ? f_user_subject.update(status: UserSubject.statuses[:finished]) : f_user_subject.update(status: UserSubject.statuses[:open])
```
Comment (vanvtt-0952) [inline app/models/process_task.rb]: ` f_user_subject.update(status: UserSubject.statuses[:open])` thừa

**Comment by duypn-2245** [comment]
ready

**Pull request merged**